alphastats/llm/llm_integration.py

```diff
@@ -355,31 +467,42 @@ def display_chat_history(self):
         None
         """
         for message in self._messages:
-            role = message["role"].capitalize()
-            content = message["content"]
-
-            if role == "Assistant" and "tool_calls" in message:
-                display(Markdown(f"**{role}**: {content}"))
-                for tool_call in message["tool_calls"]:
+            role = message[MessageKeys.ROLE]
+            content = message[MessageKeys.CONTENT]
+            tokens = self.estimate_tokens([message])
+
+            if role == Roles.ASSISTANT and MessageKeys.TOOL_CALLS in message:
+                display(
+                    Markdown(
+                        f"**{role.capitalize()}**: {content} *({str(tokens)} tokens)*"
+                    )
+                )
+                for tool_call in message[MessageKeys.TOOL_CALLS]:
                     function_name = tool_call.function.name
                     function_args = tool_call.function.arguments
                     display(Markdown(f"*Function Call*: `{function_name}`"))
                     display(Markdown(f"*Arguments*: ```json\n{function_args}\n```"))
 
             elif role == "Tool":
```

Comment:
is this different from the following?
```suggestion
            elif role == Roles.TOOL
```

Reply:
Ah, yes it's different, thanks for spotting this.